scripts/buyAuctionLands.js

```diff
@@ -0,0 +1,236 @@
+            )
+            if (!shouldBuy) process.exit()
+          }
+
+          await bidOnParcels(parcels, account, options.manaTokenAddress, {
+            txDelay: options.txDelay,
+            retryFailedTxs: options.retryFailedTxs
+          })
+        })
+      )
+  }
+}
+
+async function getValidParcels(allParcels) {
+  const parcels = []
+  const landRegistryContract = eth.getContract('LANDRegistry')
+
+  for (const parcel of allParcels) {
+    const { x, y } = parcel
+    const parcelId = buildCoordinate(x, y)
+
+    if (!Bounds.inBounds(x, y)) {
+      log.info(`Parcel ${parcelId} is out of bounds`)
+      continue
+    }
+
+    const owner = await landRegistryContract.ownerOfLand(x, y)
+    if (!Contract.isEmptyAddress(owner)) {
+      log.info(
+        `Parcel ${parcelId} already has an owner on the blockchain: ${owner}`
+      )
+      continue
+    }
+    parcels.push(parcel)
+  }
+
+  return parcels
+}
+
+async function bidOnParcels(...args) {
+  const [parcels, account, tokenAddress, { shouldRetry, txDelay }] = args
+
+  const landAuctionContract = eth.getContract('LANDAuction')
+  const gasPrice = (await landAuctionContract.gasPriceLimit()).toNumber()
+  const landsLimit = (await landAuctionContract.landsLimitPerBid()).toNumber()
+
+  if (gasPrice === 0 || landsLimit === 0) {
+    log.info(
+      'Either gasPrice or landsLimit is 0, waiting and retrying in 10 seconds'
+    )
+    await utils.sleep(10000)
+    return bidOnParcels(...args)
+  } else {
+    log.info(
+      `Using ${gasPrice} as gasPrice and ${landsLimit} as land limit per bid`
+    )
+  }
+
+  const txs = []
+  const parcelsToRetry = []
+
+  log.info(`Buying ${parcels.length} parcels with token ${tokenAddress}`)
+  await asyncBatch({
+    elements: parcels,
+    callback: async parcelsBatch => {
+      const { xs, ys } = splitCoodinatePairs(parcelsBatch)
+
+      log.info(`Sending tx for ${xs.length} parcels`)
+      const hash = await landAuctionContract.bid(
+        xs,
+        ys,
+        account,
+        tokenAddress,
+        { gasPrice: gasPrice, from: account }
+      )
+      log.info(`Got tx hash ${hash} for bidding on ${xs.length} parcels`)
+      txs.push({ hash, xs, ys })
+
+      log.info(`Sleeping ${txDelay / 1000} seconds`)
+      await utils.sleep(txDelay)
+    },
+    batchSize: landsLimit
+  })
+
+  for (const tx of txs) {
+    const { hash, xs, ys } = tx
+    log.info(`Waiting for tx: ${hash} which buys ${xs.length} parcels`)
+    const transaction = await getConfirmedTransaction(hash)
+
+    if (transaction === null) {
+      for (const [index, x] of xs.entries()) {
+        parcelsToRetry.push({ x, y: ys[index] })
+      }
+    }
+  }
+
+  log.info(`Bought ${parcels.length - parcelsToRetry.length} parcels`)
+
+  if (parcelsToRetry.length > 0 && shouldRetry) {
+    log.info(`Retrying on ${parcelsToRetry.length} parcels`)
+    return bidOnParcels(...args)
```

Comment:
isn't this retrying with the same parcels as before, instead of using only the parcels in  `parcelsToRetry`?